connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -252,6 +252,18 @@ class Dataset[T] private[sql] (
     }
   }
 
+  private def containsUnboudRowEncoder(enc: AgnosticEncoder[_]): Boolean = {
+    if (enc == UnboundRowEncoder) {
+      true
+    } else {
+      enc match {
+        case ProductEncoder(clsTag, fields) if ProductEncoder.isTuple(clsTag) =>
```

Comment:
In theory any 'struct' encoder can contain an UnboundRowEncoder. This only deals with the stuff we can currently create in scala client.